### PR TITLE
FEATURE: FET-233 Upgrade react-intl

### PR DESCRIPTION
**Pull request by tomassatinsky**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2612
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2444

# Related Jira tasks

### CODE REVIEW

**Comment by tomassatinsky** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5367/ : FAILURE in 14m 07s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/852/ : SUCCESS in 11s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2793/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/853/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2224/ : UNSTABLE in 15m 45s

File: package.json [hunk 1]
```diff
@@ -128,14 +129,15 @@
     "hoist-non-react-statics": "^3.3.0",
     "http-status-codes": "^1.3.0",
     "immer": "^1.3.1",
+    "intl-messageformat": "^7.5.4",
```
Comment (pbenes) [inline package.json]: Not sure if this has to be in dependencies... It seems to me that we're using only a type from that https://github.com/gooddata/gooddata-react-components/pull/1265/files#diff-0e397ad7f5ed5798ab35afb129ad30c8R14
Is it really needed or could it be moved to dev deps?

**Comment by tomassatinsky** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1265.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/857/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2227/ : SUCCESS in 27m 10s

**Comment by tomassatinsky** [comment]
ok to test

**Comment by tomassatinsky** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5371/ : FAILURE in 7m 30s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/860/ : SUCCESS in 11s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2797/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/861/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2229/ : UNSTABLE in 16m 01s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5372/ : SUCCESS in 10m 00s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/862/ : SUCCESS in 8s

**Comment by tomassatinsky** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5373/ : SUCCESS in 6m 34s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/863/ : SUCCESS in 11s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2798/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/864/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2230/ : SUCCESS in 15m 46s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5386/ : SUCCESS in 10m 30s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/880/ : SUCCESS in 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5387/ : SUCCESS in 6m 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/881/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5389/ : SUCCESS in 19m 50s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/884/ : SUCCESS in 28s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5422/ : SUCCESS in 10m 34s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/917/ : SUCCESS in 7s

**Comment by ivan-nejezchleb** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5423/ : SUCCESS in 12m 13s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/918/ : SUCCESS in 12s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5424/ : SUCCESS in 4m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/919/ : SUCCESS in 9s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5425/ : SUCCESS in 5m 05s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/920/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5426/ : SUCCESS in 11m 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/921/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5428/ : SUCCESS in 10m 39s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/923/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5429/ : SUCCESS in 5m 39s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/924/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5430/ : SUCCESS in 13m 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/925/ : SUCCESS in 7s

File: tsconfig.json [hunk 2]
```diff
@@ -5,6 +5,9 @@
     "jsx": "react",
     "lib": [
       "es7",
+      "ES2017.Intl",
+      "ES2018.Intl",
+      "ESNext.Intl",
```
[the remark this answers is not in the document]
Comment (ivan-nejezchleb) [reply tsconfig.json]: all three are necessary

File: package.json (hunk 1 above)
Comment (ivan-nejezchleb) [reply package.json]: moved

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5432/ : SUCCESS in 17m 08s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/927/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5436/ : SUCCESS in 9m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/933/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5440/ : SUCCESS in 19m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/938/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5492/ : FAILURE in 4m 11s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/990/ : SUCCESS in 5s

File: src/components/core/base/VisualizationLoadingHOC.tsx [hunk 3]
```diff
@@ -426,17 +432,5 @@ export function visualizationLoadingHOC<
                 return Promise.reject(error);
             };
         }
-    }
-
-    const IntlLoadingHOC = injectIntl(LoadingHOCWrapped);
```
[the remark this answers is not in the document]
Comment (ivan-nejezchleb) [reply src/components/core/base/VisualizationLoadingHOC.tsx]: rewritten back to injectIntl

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5493/ : SUCCESS in 9m 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/991/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2838/ : SUCCESS in 15m 22s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5494/ : SUCCESS in 4m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1339/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1835/ : SUCCESS in 20m 01s

**Pull request merged**